src/example/org/deidentifier/arx/examples/Example59.java

```diff
@@ -1,44 +1,86 @@
 package org.deidentifier.arx.examples;
 
-import org.deidentifier.arx.framework.data.DataColumn;
-import org.deidentifier.arx.framework.data.DataMatrix;
-import org.deidentifier.arx.framework.data.Dictionary;
-import org.deidentifier.arx.masking.DataMaskingFunction.PermutationFunctionColumns;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Iterator;
 
-public class Example59 {
-	
-	public static void main (String[] args) {
-		int rows = 1000;
-		int columns = 2;
-		
-		DataMatrix data = new DataMatrix(rows, columns);
-		String[] header = new String[]{"column1","column2"};
-		int column = 0;
-		Dictionary dictionary = new Dictionary(rows);
+import org.deidentifier.arx.ARXAnonymizer;
+import org.deidentifier.arx.ARXConfiguration;
+import org.deidentifier.arx.ARXResult;
+import org.deidentifier.arx.AttributeType;
+import org.deidentifier.arx.Data;
+import org.deidentifier.arx.AttributeType.Hierarchy;
+import org.deidentifier.arx.AttributeType.MaskingFunction;
+import org.deidentifier.arx.AttributeType.Hierarchy.DefaultHierarchy;
+import org.deidentifier.arx.Data.DefaultData;
+import org.deidentifier.arx.criteria.KAnonymity;
 
-		int[] tuple = new int[columns];
-		
-		for (int i=0;i<rows;i++) {
-			
-			for(int j=0;j<columns;j++) {
-				tuple[j] = dictionary.register(j, Integer.toString(j+1) + "_ColumnStringTest_" + Integer.toString(i+1)+"row");
-			}
-			data.setRow(i, tuple);
-		}
-		
-		
-		dictionary.finalizeAll();
-		DataColumn c = new DataColumn(data, header, column, dictionary);
-		
-		PermutationFunctionColumns pfc = new PermutationFunctionColumns(true);
-		for (int i=0; i<c.getNumRows();i++) {
-			System.out.println(c.get(i));
-		}
-		System.out.println();
-		pfc.apply(c);
+/**
+ * This class implements an example of permutation function on columns
+ * 
+ * @author giupardeb
+ *
+ */
+public class Example59 extends Example {
+	/**
+	 * Entry point.
+	 * 
+	 * @param args the arguments
+	 * @throws IOException
+	 */
+	public static void main (String[] args) throws IOException {
 		
-		for (int i=0; i<c.getNumRows();i++) {
-			System.out.println(c.get(i));
-		}
+		// Define data
+        DefaultData data = Data.create();
+        data.add("id","Name", "Surname");
+        data.add("1","Gerek", "Macourek");
+        data.add("2","Nadia", "Stare");
+        data.add("3","Bobby", "Spera");
+        data.add("4","Carly", "Avrahamof");
+        data.add("5","Morgan", "MacCaughey");
+        data.add("6","Leilah", "Yapp");
+        data.add("7","Alida", "Stud");
+        data.add("8","Shannon", "Diwell");
+        data.add("9","Kaitlin", "Farmar");
+        data.add("10","Angela", "Pinkett");
+        
+        
+        DefaultHierarchy id = Hierarchy.create();
```

Comment:
You can just mark the column "id" as identifying. This will result in all values being replaced with "*".

Reply:
I marked the column "id" as identifying:
`data.getDefinition().setAttributeType("id", AttributeType.IDENTIFYING_ATTRIBUTE);`

Then I marked columns Name and Surname as quay Identifer:
```
data.getDefinition().setAttributeType("Name",AttributeType.QUASI_IDENTIFYING_ATTRIBUTE);
data.getDefinition().setAttributeType("Surname",AttributeType.QUASI_IDENTIFYING_ATTRIBUTE);
```
but when I run the Example, this is the output:

```
 - Time needed: 0.02s
Exception in thread "main" java.lang.NullPointerException
	at org.deidentifier.arx.examples.Example59.main(Example59.java:85)
 - No solution found!
 - Transformed data:

```

Where am I wrong?

Reply:
Try: config.setSuppressionLimit(1d);

Reply:
Okay, I will use the original configuration. Don't worry, thank you so much for your availability!